{
 "cells": [
  {
   "cell_type": "markdown",
   "metadata": {
    "id": "ifhgKXirU9Ds"
   },
   "source": [
    "# I] Import and download"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 1,
   "metadata": {
    "id": "TLQwwG5VDPFA"
   },
   "outputs": [],
   "source": [
    "import pandas"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 2,
   "metadata": {},
   "outputs": [],
   "source": [
    "input_file_directory = '/Users/kja11/OneDrive - Háskóli Íslands/PhD ATG7//0 in_silico/Python/1)data_input/'\n",
    "output_file_directory = '/Users/kja11/OneDrive - Háskóli Íslands/PhD ATG7/0 in_silico/Python/3)output/'"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 13,
   "metadata": {},
   "outputs": [],
   "source": [
    "path = output_file_directory + \"ATG7/TCGA_GTEX/all_tissues/\"\n",
    "path2 = 'dataframes_for_input/'"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 8,
   "metadata": {},
   "outputs": [
    {
     "name": "stdout",
     "output_type": "stream",
     "text": [
      "(5847, 17)\n",
      "(8022, 17)\n",
      "(13869, 17)\n",
      "Wall time: 64.8 ms\n"
     ]
    },
    {
     "data": {
      "text/html": [
       "<div>\n",
       "<style scoped>\n",
       "    .dataframe tbody tr th:only-of-type {\n",
       "        vertical-align: middle;\n",
       "    }\n",
       "\n",
       "    .dataframe tbody tr th {\n",
       "        vertical-align: top;\n",
       "    }\n",
       "\n",
       "    .dataframe thead th {\n",
       "        text-align: right;\n",
       "    }\n",
       "</style>\n",
       "<table border=\"1\" class=\"dataframe\">\n",
       "  <thead>\n",
       "    <tr style=\"text-align: right;\">\n",
       "      <th></th>\n",
       "      <th>sample</th>\n",
       "      <th>Sample_Type</th>\n",
       "      <th>Primary_Site</th>\n",
       "      <th>ATG7</th>\n",
       "      <th>ATG7_1</th>\n",
       "      <th>ATG7_2</th>\n",
       "      <th>ATG7_3</th>\n",
       "      <th>ATG7_tpm</th>\n",
       "      <th>ATG7_1tpm</th>\n",
       "      <th>ATG7_2tpm</th>\n",
       "      <th>ATG7_3tpm</th>\n",
       "      <th>ATG7_prot_tot</th>\n",
       "      <th>ATG7_1perc</th>\n",
       "      <th>ATG7_2perc</th>\n",
       "      <th>ATG7_3perc</th>\n",
       "      <th>ATG7_1norm</th>\n",
       "      <th>ATG7_2norm</th>\n",
       "    </tr>\n",
       "  </thead>\n",
       "  <tbody>\n",
       "    <tr>\n",
       "      <th>0</th>\n",
       "      <td>GTEX-ZTTD-0326-SM-57WFW</td>\n",
       "      <td>Normal Tissue</td>\n",
       "      <td>Muscle</td>\n",
       "      <td>5.085097</td>\n",
       "      <td>3.955992</td>\n",
       "      <td>2.641679</td>\n",
       "      <td>1.137484</td>\n",
       "      <td>32.944283</td>\n",
       "      <td>14.519306</td>\n",
       "      <td>5.240574</td>\n",
       "      <td>1.19997</td>\n",
       "      <td>20.959850</td>\n",
       "      <td>69.271996</td>\n",
       "      <td>25.002916</td>\n",
       "      <td>5.725088</td>\n",
       "      <td>0.440723</td>\n",
       "      <td>0.159074</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>1</th>\n",
       "      <td>GTEX-OXRO-1726-SM-3LK6C</td>\n",
       "      <td>Normal Tissue</td>\n",
       "      <td>Muscle</td>\n",
       "      <td>4.001888</td>\n",
       "      <td>2.821767</td>\n",
       "      <td>1.823465</td>\n",
       "      <td>0.000000</td>\n",
       "      <td>15.020948</td>\n",
       "      <td>6.070277</td>\n",
       "      <td>2.539302</td>\n",
       "      <td>0.00000</td>\n",
       "      <td>8.609579</td>\n",
       "      <td>70.506084</td>\n",
       "      <td>29.493916</td>\n",
       "      <td>0.000000</td>\n",
       "      <td>0.404121</td>\n",
       "      <td>0.169051</td>\n",
       "    </tr>\n",
       "  </tbody>\n",
       "</table>\n",
       "</div>"
      ],
      "text/plain": [
       "                    sample    Sample_Type Primary_Site      ATG7    ATG7_1  \\\n",
       "0  GTEX-ZTTD-0326-SM-57WFW  Normal Tissue       Muscle  5.085097  3.955992   \n",
       "1  GTEX-OXRO-1726-SM-3LK6C  Normal Tissue       Muscle  4.001888  2.821767   \n",
       "\n",
       "     ATG7_2    ATG7_3   ATG7_tpm  ATG7_1tpm  ATG7_2tpm  ATG7_3tpm  \\\n",
       "0  2.641679  1.137484  32.944283  14.519306   5.240574    1.19997   \n",
       "1  1.823465  0.000000  15.020948   6.070277   2.539302    0.00000   \n",
       "\n",
       "   ATG7_prot_tot  ATG7_1perc  ATG7_2perc  ATG7_3perc  ATG7_1norm  ATG7_2norm  \n",
       "0      20.959850   69.271996   25.002916    5.725088    0.440723    0.159074  \n",
       "1       8.609579   70.506084   29.493916    0.000000    0.404121    0.169051  "
      ]
     },
     "execution_count": 8,
     "metadata": {},
     "output_type": "execute_result"
    }
   ],
   "source": [
    "%%time\n",
    "# DL data ATG7 isoform expression\n",
    "df_ATG7_Normal = pandas.read_csv(path+path2+'df_ATG7_Normal_all.csv', sep = \",\")\n",
    "df_ATG7_Primary = pandas.read_csv(path+path2+'df_ATG7_Primary_all.csv', sep = \",\")\n",
    "\n",
    "#remove Survival information\n",
    "dfs = df_ATG7_Normal, df_ATG7_Primary\n",
    "for df in dfs:\n",
    "    df.drop(df.columns[[3, 4]], axis = 1, inplace = True)\n",
    "\n",
    "print(df_ATG7_Normal.shape)\n",
    "print(df_ATG7_Primary.shape)   \n",
    "\n",
    "#create df with the two groups\n",
    "df_ATG7_NormPrim = pandas.concat([df_ATG7_Normal, df_ATG7_Primary], axis=0)\n",
    "print(df_ATG7_NormPrim.shape)\n",
    "df_ATG7_NormPrim.head(2)"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 9,
   "metadata": {},
   "outputs": [
    {
     "name": "stdout",
     "output_type": "stream",
     "text": [
      "(7429, 19383)\n",
      "(9185, 19383)\n",
      "Wall time: 3min 52s\n"
     ]
    },
    {
     "data": {
      "text/html": [
       "<div>\n",
       "<style scoped>\n",
       "    .dataframe tbody tr th:only-of-type {\n",
       "        vertical-align: middle;\n",
       "    }\n",
       "\n",
       "    .dataframe tbody tr th {\n",
       "        vertical-align: top;\n",
       "    }\n",
       "\n",
       "    .dataframe thead th {\n",
       "        text-align: right;\n",
       "    }\n",
       "</style>\n",
       "<table border=\"1\" class=\"dataframe\">\n",
       "  <thead>\n",
       "    <tr style=\"text-align: right;\">\n",
       "      <th></th>\n",
       "      <th>sample</th>\n",
       "      <th>ENSG00000198888</th>\n",
       "      <th>ENSG00000198763</th>\n",
       "      <th>ENSG00000198804</th>\n",
       "      <th>ENSG00000198712</th>\n",
       "      <th>ENSG00000228253</th>\n",
       "      <th>ENSG00000198899</th>\n",
       "      <th>ENSG00000198938</th>\n",
       "      <th>ENSG00000198840</th>\n",
       "      <th>ENSG00000212907</th>\n",
       "      <th>...</th>\n",
       "      <th>ENSG00000160678</th>\n",
       "      <th>ENSG00000160679</th>\n",
       "      <th>ENSG00000143553</th>\n",
       "      <th>ENSG00000214193</th>\n",
       "      <th>ENSG00000196182</th>\n",
       "      <th>ENSG00000181817</th>\n",
       "      <th>ENSG00000116885</th>\n",
       "      <th>ENSG00000116898</th>\n",
       "      <th>ENSG00000119535</th>\n",
       "      <th>ENSG00000142694</th>\n",
       "    </tr>\n",
       "  </thead>\n",
       "  <tbody>\n",
       "    <tr>\n",
       "      <th>0</th>\n",
       "      <td>TCGA-02-0047-01</td>\n",
       "      <td>12.920386</td>\n",
       "      <td>13.122162</td>\n",
       "      <td>14.013687</td>\n",
       "      <td>13.525222</td>\n",
       "      <td>13.843998</td>\n",
       "      <td>13.777803</td>\n",
       "      <td>13.403833</td>\n",
       "      <td>12.812500</td>\n",
       "      <td>13.418832</td>\n",
       "      <td>...</td>\n",
       "      <td>6.490222</td>\n",
       "      <td>5.827309</td>\n",
       "      <td>6.513364</td>\n",
       "      <td>2.443619</td>\n",
       "      <td>5.282026</td>\n",
       "      <td>6.072280</td>\n",
       "      <td>3.985485</td>\n",
       "      <td>5.569572</td>\n",
       "      <td>4.365281</td>\n",
       "      <td>4.724089</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>1</th>\n",
       "      <td>TCGA-02-0055-01</td>\n",
       "      <td>10.398268</td>\n",
       "      <td>10.778221</td>\n",
       "      <td>12.394768</td>\n",
       "      <td>12.389569</td>\n",
       "      <td>11.728425</td>\n",
       "      <td>11.941167</td>\n",
       "      <td>12.202506</td>\n",
       "      <td>11.014897</td>\n",
       "      <td>11.022793</td>\n",
       "      <td>...</td>\n",
       "      <td>6.019293</td>\n",
       "      <td>5.716474</td>\n",
       "      <td>6.084693</td>\n",
       "      <td>2.307414</td>\n",
       "      <td>4.399842</td>\n",
       "      <td>6.515045</td>\n",
       "      <td>3.860966</td>\n",
       "      <td>6.722909</td>\n",
       "      <td>4.654781</td>\n",
       "      <td>5.545689</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>2</th>\n",
       "      <td>TCGA-02-2483-01</td>\n",
       "      <td>11.694535</td>\n",
       "      <td>12.025246</td>\n",
       "      <td>13.115462</td>\n",
       "      <td>13.408733</td>\n",
       "      <td>13.628814</td>\n",
       "      <td>13.422831</td>\n",
       "      <td>13.443729</td>\n",
       "      <td>12.998576</td>\n",
       "      <td>11.940967</td>\n",
       "      <td>...</td>\n",
       "      <td>4.865893</td>\n",
       "      <td>6.380602</td>\n",
       "      <td>6.655963</td>\n",
       "      <td>1.035643</td>\n",
       "      <td>4.701003</td>\n",
       "      <td>6.079570</td>\n",
       "      <td>3.215637</td>\n",
       "      <td>6.306130</td>\n",
       "      <td>3.625293</td>\n",
       "      <td>4.073828</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>3</th>\n",
       "      <td>TCGA-02-2485-01</td>\n",
       "      <td>10.430545</td>\n",
       "      <td>10.518083</td>\n",
       "      <td>11.979057</td>\n",
       "      <td>11.614360</td>\n",
       "      <td>12.047740</td>\n",
       "      <td>11.772712</td>\n",
       "      <td>11.765314</td>\n",
       "      <td>10.914447</td>\n",
       "      <td>10.930739</td>\n",
       "      <td>...</td>\n",
       "      <td>5.138313</td>\n",
       "      <td>6.309783</td>\n",
       "      <td>6.574205</td>\n",
       "      <td>1.130945</td>\n",
       "      <td>4.563191</td>\n",
       "      <td>6.101149</td>\n",
       "      <td>3.840955</td>\n",
       "      <td>6.224800</td>\n",
       "      <td>2.825804</td>\n",
       "      <td>3.810465</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>4</th>\n",
       "      <td>TCGA-04-1331-01</td>\n",
       "      <td>14.069484</td>\n",
       "      <td>14.629057</td>\n",
       "      <td>14.281572</td>\n",
       "      <td>14.285772</td>\n",
       "      <td>13.898994</td>\n",
       "      <td>14.339270</td>\n",
       "      <td>14.284872</td>\n",
       "      <td>13.664611</td>\n",
       "      <td>13.569319</td>\n",
       "      <td>...</td>\n",
       "      <td>8.902714</td>\n",
       "      <td>6.118692</td>\n",
       "      <td>5.581614</td>\n",
       "      <td>2.137535</td>\n",
       "      <td>4.390222</td>\n",
       "      <td>5.167957</td>\n",
       "      <td>3.933549</td>\n",
       "      <td>6.779479</td>\n",
       "      <td>2.042638</td>\n",
       "      <td>5.339833</td>\n",
       "    </tr>\n",
       "  </tbody>\n",
       "</table>\n",
       "<p>5 rows × 19383 columns</p>\n",
       "</div>"
      ],
      "text/plain": [
       "            sample  ENSG00000198888  ENSG00000198763  ENSG00000198804  \\\n",
       "0  TCGA-02-0047-01        12.920386        13.122162        14.013687   \n",
       "1  TCGA-02-0055-01        10.398268        10.778221        12.394768   \n",
       "2  TCGA-02-2483-01        11.694535        12.025246        13.115462   \n",
       "3  TCGA-02-2485-01        10.430545        10.518083        11.979057   \n",
       "4  TCGA-04-1331-01        14.069484        14.629057        14.281572   \n",
       "\n",
       "   ENSG00000198712  ENSG00000228253  ENSG00000198899  ENSG00000198938  \\\n",
       "0        13.525222        13.843998        13.777803        13.403833   \n",
       "1        12.389569        11.728425        11.941167        12.202506   \n",
       "2        13.408733        13.628814        13.422831        13.443729   \n",
       "3        11.614360        12.047740        11.772712        11.765314   \n",
       "4        14.285772        13.898994        14.339270        14.284872   \n",
       "\n",
       "   ENSG00000198840  ENSG00000212907  ...  ENSG00000160678  ENSG00000160679  \\\n",
       "0        12.812500        13.418832  ...         6.490222         5.827309   \n",
       "1        11.014897        11.022793  ...         6.019293         5.716474   \n",
       "2        12.998576        11.940967  ...         4.865893         6.380602   \n",
       "3        10.914447        10.930739  ...         5.138313         6.309783   \n",
       "4        13.664611        13.569319  ...         8.902714         6.118692   \n",
       "\n",
       "   ENSG00000143553  ENSG00000214193  ENSG00000196182  ENSG00000181817  \\\n",
       "0         6.513364         2.443619         5.282026         6.072280   \n",
       "1         6.084693         2.307414         4.399842         6.515045   \n",
       "2         6.655963         1.035643         4.701003         6.079570   \n",
       "3         6.574205         1.130945         4.563191         6.101149   \n",
       "4         5.581614         2.137535         4.390222         5.167957   \n",
       "\n",
       "   ENSG00000116885  ENSG00000116898  ENSG00000119535  ENSG00000142694  \n",
       "0         3.985485         5.569572         4.365281         4.724089  \n",
       "1         3.860966         6.722909         4.654781         5.545689  \n",
       "2         3.215637         6.306130         3.625293         4.073828  \n",
       "3         3.840955         6.224800         2.825804         3.810465  \n",
       "4         3.933549         6.779479         2.042638         5.339833  \n",
       "\n",
       "[5 rows x 19383 columns]"
      ]
     },
     "execution_count": 9,
     "metadata": {},
     "output_type": "execute_result"
    }
   ],
   "source": [
    "%%time\n",
    "# DL data gene expression for Normal Tissue and Primary Tumors\n",
    "path = output_file_directory + \"dataframes_for_input/\"\n",
    "ensembl_normal_protcoding = pandas.read_csv(path+\"ensembl_normal_protcoding_expr.tsv\", sep = \"\\t\")\n",
    "ensembl_primary_protcoding = pandas.read_csv(path+\"ensembl_primary_protcoding_expr.tsv\", sep = \"\\t\")\n",
    "\n",
    "print(ensembl_normal_protcoding.shape)\n",
    "print(ensembl_primary_protcoding.shape)\n",
    "\n",
    "ensembl_primary_protcoding.head()"
   ]
  },
  {
   "cell_type": "markdown",
   "metadata": {},
   "source": [
    "# II] ATG7 isoforms + Whole gene expression (ensembl)"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 10,
   "metadata": {
    "id": "9b0Bg43gEMRM",
    "scrolled": true
   },
   "outputs": [
    {
     "name": "stdout",
     "output_type": "stream",
     "text": [
      "(5847, 19384)\n",
      "(8022, 19384)\n"
     ]
    },
    {
     "data": {
      "text/html": [
       "<div>\n",
       "<style scoped>\n",
       "    .dataframe tbody tr th:only-of-type {\n",
       "        vertical-align: middle;\n",
       "    }\n",
       "\n",
       "    .dataframe tbody tr th {\n",
       "        vertical-align: top;\n",
       "    }\n",
       "\n",
       "    .dataframe thead th {\n",
       "        text-align: right;\n",
       "    }\n",
       "</style>\n",
       "<table border=\"1\" class=\"dataframe\">\n",
       "  <thead>\n",
       "    <tr style=\"text-align: right;\">\n",
       "      <th></th>\n",
       "      <th>Sample_Type</th>\n",
       "      <th>Primary_Site</th>\n",
       "      <th>ATG7</th>\n",
       "      <th>ATG7_1</th>\n",
       "      <th>ATG7_2</th>\n",
       "      <th>ENSG00000198712</th>\n",
       "      <th>ENSG00000228253</th>\n",
       "      <th>ENSG00000198899</th>\n",
       "      <th>ENSG00000198938</th>\n",
       "      <th>ENSG00000198840</th>\n",
       "      <th>...</th>\n",
       "      <th>ENSG00000160678</th>\n",
       "      <th>ENSG00000160679</th>\n",
       "      <th>ENSG00000143553</th>\n",
       "      <th>ENSG00000214193</th>\n",
       "      <th>ENSG00000196182</th>\n",
       "      <th>ENSG00000181817</th>\n",
       "      <th>ENSG00000116885</th>\n",
       "      <th>ENSG00000116898</th>\n",
       "      <th>ENSG00000119535</th>\n",
       "      <th>ENSG00000142694</th>\n",
       "    </tr>\n",
       "  </thead>\n",
       "  <tbody>\n",
       "    <tr>\n",
       "      <th>0</th>\n",
       "      <td>Normal Tissue</td>\n",
       "      <td>Muscle</td>\n",
       "      <td>5.085097</td>\n",
       "      <td>3.955992</td>\n",
       "      <td>2.641679</td>\n",
       "      <td>14.445365</td>\n",
       "      <td>15.097141</td>\n",
       "      <td>14.472063</td>\n",
       "      <td>14.768352</td>\n",
       "      <td>13.814400</td>\n",
       "      <td>...</td>\n",
       "      <td>2.981848</td>\n",
       "      <td>5.677832</td>\n",
       "      <td>5.063070</td>\n",
       "      <td>2.901118</td>\n",
       "      <td>4.588581</td>\n",
       "      <td>5.141228</td>\n",
       "      <td>2.601721</td>\n",
       "      <td>6.100164</td>\n",
       "      <td>6.502089</td>\n",
       "      <td>5.015268</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>1</th>\n",
       "      <td>Normal Tissue</td>\n",
       "      <td>Muscle</td>\n",
       "      <td>4.001888</td>\n",
       "      <td>2.821767</td>\n",
       "      <td>1.823465</td>\n",
       "      <td>13.821200</td>\n",
       "      <td>14.816150</td>\n",
       "      <td>14.110781</td>\n",
       "      <td>14.703554</td>\n",
       "      <td>13.277745</td>\n",
       "      <td>...</td>\n",
       "      <td>9.067289</td>\n",
       "      <td>5.997733</td>\n",
       "      <td>5.158236</td>\n",
       "      <td>2.217237</td>\n",
       "      <td>4.671302</td>\n",
       "      <td>5.053079</td>\n",
       "      <td>2.370170</td>\n",
       "      <td>5.810708</td>\n",
       "      <td>2.124335</td>\n",
       "      <td>4.707639</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>2</th>\n",
       "      <td>Normal Tissue</td>\n",
       "      <td>Muscle</td>\n",
       "      <td>3.950379</td>\n",
       "      <td>2.792620</td>\n",
       "      <td>1.883517</td>\n",
       "      <td>14.062284</td>\n",
       "      <td>15.589729</td>\n",
       "      <td>14.877448</td>\n",
       "      <td>14.545960</td>\n",
       "      <td>13.778903</td>\n",
       "      <td>...</td>\n",
       "      <td>9.227606</td>\n",
       "      <td>5.525729</td>\n",
       "      <td>5.617656</td>\n",
       "      <td>0.790754</td>\n",
       "      <td>5.691266</td>\n",
       "      <td>6.396905</td>\n",
       "      <td>2.792876</td>\n",
       "      <td>7.330383</td>\n",
       "      <td>2.021477</td>\n",
       "      <td>1.978191</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>3</th>\n",
       "      <td>Normal Tissue</td>\n",
       "      <td>Muscle</td>\n",
       "      <td>3.756766</td>\n",
       "      <td>2.352930</td>\n",
       "      <td>1.589822</td>\n",
       "      <td>14.152279</td>\n",
       "      <td>15.603129</td>\n",
       "      <td>14.957645</td>\n",
       "      <td>14.132380</td>\n",
       "      <td>13.885395</td>\n",
       "      <td>...</td>\n",
       "      <td>9.563806</td>\n",
       "      <td>6.063118</td>\n",
       "      <td>5.653030</td>\n",
       "      <td>0.948623</td>\n",
       "      <td>7.323129</td>\n",
       "      <td>6.529981</td>\n",
       "      <td>2.053104</td>\n",
       "      <td>6.799101</td>\n",
       "      <td>2.277958</td>\n",
       "      <td>2.192172</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>4</th>\n",
       "      <td>Normal Tissue</td>\n",
       "      <td>Muscle</td>\n",
       "      <td>3.704980</td>\n",
       "      <td>2.540744</td>\n",
       "      <td>2.295295</td>\n",
       "      <td>14.262573</td>\n",
       "      <td>15.421933</td>\n",
       "      <td>14.876148</td>\n",
       "      <td>14.429965</td>\n",
       "      <td>13.765204</td>\n",
       "      <td>...</td>\n",
       "      <td>8.513948</td>\n",
       "      <td>6.009251</td>\n",
       "      <td>5.306292</td>\n",
       "      <td>1.104343</td>\n",
       "      <td>5.907119</td>\n",
       "      <td>5.832418</td>\n",
       "      <td>2.185847</td>\n",
       "      <td>6.858078</td>\n",
       "      <td>1.735488</td>\n",
       "      <td>2.761269</td>\n",
       "    </tr>\n",
       "  </tbody>\n",
       "</table>\n",
       "<p>5 rows × 19384 columns</p>\n",
       "</div>"
      ],
      "text/plain": [
       "     Sample_Type Primary_Site      ATG7    ATG7_1    ATG7_2  ENSG00000198712  \\\n",
       "0  Normal Tissue       Muscle  5.085097  3.955992  2.641679        14.445365   \n",
       "1  Normal Tissue       Muscle  4.001888  2.821767  1.823465        13.821200   \n",
       "2  Normal Tissue       Muscle  3.950379  2.792620  1.883517        14.062284   \n",
       "3  Normal Tissue       Muscle  3.756766  2.352930  1.589822        14.152279   \n",
       "4  Normal Tissue       Muscle  3.704980  2.540744  2.295295        14.262573   \n",
       "\n",
       "   ENSG00000228253  ENSG00000198899  ENSG00000198938  ENSG00000198840  ...  \\\n",
       "0        15.097141        14.472063        14.768352        13.814400  ...   \n",
       "1        14.816150        14.110781        14.703554        13.277745  ...   \n",
       "2        15.589729        14.877448        14.545960        13.778903  ...   \n",
       "3        15.603129        14.957645        14.132380        13.885395  ...   \n",
       "4        15.421933        14.876148        14.429965        13.765204  ...   \n",
       "\n",
       "   ENSG00000160678  ENSG00000160679  ENSG00000143553  ENSG00000214193  \\\n",
       "0         2.981848         5.677832         5.063070         2.901118   \n",
       "1         9.067289         5.997733         5.158236         2.217237   \n",
       "2         9.227606         5.525729         5.617656         0.790754   \n",
       "3         9.563806         6.063118         5.653030         0.948623   \n",
       "4         8.513948         6.009251         5.306292         1.104343   \n",
       "\n",
       "   ENSG00000196182  ENSG00000181817  ENSG00000116885  ENSG00000116898  \\\n",
       "0         4.588581         5.141228         2.601721         6.100164   \n",
       "1         4.671302         5.053079         2.370170         5.810708   \n",
       "2         5.691266         6.396905         2.792876         7.330383   \n",
       "3         7.323129         6.529981         2.053104         6.799101   \n",
       "4         5.907119         5.832418         2.185847         6.858078   \n",
       "\n",
       "   ENSG00000119535  ENSG00000142694  \n",
       "0         6.502089         5.015268  \n",
       "1         2.124335         4.707639  \n",
       "2         2.021477         1.978191  \n",
       "3         2.277958         2.192172  \n",
       "4         1.735488         2.761269  \n",
       "\n",
       "[5 rows x 19384 columns]"
      ]
     },
     "execution_count": 10,
     "metadata": {},
     "output_type": "execute_result"
    }
   ],
   "source": [
    "#Preparation of the global dataframe with ATG7 isoform expression AND whole gene expression\n",
    "data = [[ensembl_normal_protcoding, df_ATG7_Normal], [ensembl_primary_protcoding, df_ATG7_Primary]]\n",
    "\n",
    "df_final = []\n",
    "\n",
    "for df in data:\n",
    "    #Merge the two df to have ATG7 isoform expression + whole gene expression \n",
    "  df_merged = pandas.merge(df[1], df[0], on= 'sample')\n",
    "\n",
    "  #delete the sample column and select column\n",
    "  df_merged = df_merged.drop(\"sample\", 1)\n",
    "  df_merged = df_merged.drop(df_merged.columns[5:19], axis=1)\n",
    "  \n",
    "  #save in list\n",
    "  df_final.append(df_merged)\n",
    "\n",
    "ensembl_normal_protcoding_atg7_expr = df_final[0]\n",
    "ensembl_primary_protcoding_atg7_expr = df_final[1]\n",
    "\n",
    "print(ensembl_normal_protcoding_atg7_expr.shape)\n",
    "print(ensembl_primary_protcoding_atg7_expr.shape)\n",
    "ensembl_normal_protcoding_atg7_expr.head()"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 14,
   "metadata": {},
   "outputs": [],
   "source": [
    "#save to csv\n",
    "ensembl_normal_protcoding_atg7_expr.to_csv(path+path2+'ensembl_normal_atg7_proteinGE.tsv',sep = \"\\t\", index=False)\n",
    "ensembl_primary_protcoding_atg7_expr.to_csv(path+path2+'ensembl_primary_atg7_proteinGE.tsv',sep = \"\\t\", index=False)"
   ]
  },
  {
   "cell_type": "markdown",
   "metadata": {},
   "source": [
    "## B) ATG7 isoforms + Whole gene expression (symbol)"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 18,
   "metadata": {},
   "outputs": [
    {
     "name": "stdout",
     "output_type": "stream",
     "text": [
      "(7429, 19190)\n",
      "(9185, 19190)\n"
     ]
    },
    {
     "data": {
      "text/html": [
       "<div>\n",
       "<style scoped>\n",
       "    .dataframe tbody tr th:only-of-type {\n",
       "        vertical-align: middle;\n",
       "    }\n",
       "\n",
       "    .dataframe tbody tr th {\n",
       "        vertical-align: top;\n",
       "    }\n",
       "\n",
       "    .dataframe thead th {\n",
       "        text-align: right;\n",
       "    }\n",
       "</style>\n",
       "<table border=\"1\" class=\"dataframe\">\n",
       "  <thead>\n",
       "    <tr style=\"text-align: right;\">\n",
       "      <th></th>\n",
       "      <th>sample</th>\n",
       "      <th>MT-ND1</th>\n",
       "      <th>MT-ND2</th>\n",
       "      <th>MT-CO1</th>\n",
       "      <th>MT-CO2</th>\n",
       "      <th>MT-ATP8</th>\n",
       "      <th>MT-ATP6</th>\n",
       "      <th>MT-CO3</th>\n",
       "      <th>MT-ND3</th>\n",
       "      <th>MT-ND4L</th>\n",
       "      <th>...</th>\n",
       "      <th>S100A1</th>\n",
       "      <th>CHTOP</th>\n",
       "      <th>SNAPIN</th>\n",
       "      <th>SH3D21</th>\n",
       "      <th>STK40</th>\n",
       "      <th>LSM10</th>\n",
       "      <th>OSCP1</th>\n",
       "      <th>MRPS15</th>\n",
       "      <th>CSF3R</th>\n",
       "      <th>EVA1B</th>\n",
       "    </tr>\n",
       "  </thead>\n",
       "  <tbody>\n",
       "    <tr>\n",
       "      <th>0</th>\n",
       "      <td>TCGA-02-0047-01</td>\n",
       "      <td>12.9202</td>\n",
       "      <td>13.1220</td>\n",
       "      <td>14.0136</td>\n",
       "      <td>13.5251</td>\n",
       "      <td>13.8439</td>\n",
       "      <td>13.7777</td>\n",
       "      <td>13.4037</td>\n",
       "      <td>12.8123</td>\n",
       "      <td>13.4187</td>\n",
       "      <td>...</td>\n",
       "      <td>6.4741</td>\n",
       "      <td>5.8017</td>\n",
       "      <td>6.4975</td>\n",
       "      <td>2.1509</td>\n",
       "      <td>5.2445</td>\n",
       "      <td>6.0507</td>\n",
       "      <td>3.8915</td>\n",
       "      <td>5.5389</td>\n",
       "      <td>4.2936</td>\n",
       "      <td>4.6685</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>1</th>\n",
       "      <td>TCGA-02-0055-01</td>\n",
       "      <td>10.3972</td>\n",
       "      <td>10.7774</td>\n",
       "      <td>12.3945</td>\n",
       "      <td>12.3893</td>\n",
       "      <td>11.7280</td>\n",
       "      <td>11.9408</td>\n",
       "      <td>12.2022</td>\n",
       "      <td>11.0142</td>\n",
       "      <td>11.0221</td>\n",
       "      <td>...</td>\n",
       "      <td>5.9969</td>\n",
       "      <td>5.6888</td>\n",
       "      <td>6.0633</td>\n",
       "      <td>1.9822</td>\n",
       "      <td>4.3299</td>\n",
       "      <td>6.4992</td>\n",
       "      <td>3.7582</td>\n",
       "      <td>6.7092</td>\n",
       "      <td>4.5964</td>\n",
       "      <td>5.5145</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>2</th>\n",
       "      <td>TCGA-02-2483-01</td>\n",
       "      <td>11.6941</td>\n",
       "      <td>12.0249</td>\n",
       "      <td>13.1153</td>\n",
       "      <td>13.4086</td>\n",
       "      <td>13.6287</td>\n",
       "      <td>13.4227</td>\n",
       "      <td>13.4436</td>\n",
       "      <td>12.9984</td>\n",
       "      <td>11.9406</td>\n",
       "      <td>...</td>\n",
       "      <td>4.8156</td>\n",
       "      <td>6.3632</td>\n",
       "      <td>6.6416</td>\n",
       "      <td>0.0718</td>\n",
       "      <td>4.6445</td>\n",
       "      <td>6.0581</td>\n",
       "      <td>3.0515</td>\n",
       "      <td>6.2878</td>\n",
       "      <td>3.5035</td>\n",
       "      <td>3.9856</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>3</th>\n",
       "      <td>TCGA-02-2485-01</td>\n",
       "      <td>10.4295</td>\n",
       "      <td>10.5171</td>\n",
       "      <td>11.9787</td>\n",
       "      <td>11.6139</td>\n",
       "      <td>12.0474</td>\n",
       "      <td>11.7723</td>\n",
       "      <td>11.7649</td>\n",
       "      <td>10.9137</td>\n",
       "      <td>10.9300</td>\n",
       "      <td>...</td>\n",
       "      <td>5.0968</td>\n",
       "      <td>6.2915</td>\n",
       "      <td>6.5590</td>\n",
       "      <td>0.2522</td>\n",
       "      <td>4.5009</td>\n",
       "      <td>6.0800</td>\n",
       "      <td>3.7367</td>\n",
       "      <td>6.2054</td>\n",
       "      <td>2.6067</td>\n",
       "      <td>3.7039</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>4</th>\n",
       "      <td>TCGA-04-1331-01</td>\n",
       "      <td>14.0694</td>\n",
       "      <td>14.6290</td>\n",
       "      <td>14.2815</td>\n",
       "      <td>14.2857</td>\n",
       "      <td>13.8989</td>\n",
       "      <td>14.3392</td>\n",
       "      <td>14.2848</td>\n",
       "      <td>13.6645</td>\n",
       "      <td>13.5692</td>\n",
       "      <td>...</td>\n",
       "      <td>8.8997</td>\n",
       "      <td>6.0978</td>\n",
       "      <td>5.5512</td>\n",
       "      <td>1.7660</td>\n",
       "      <td>4.3198</td>\n",
       "      <td>5.1273</td>\n",
       "      <td>3.8360</td>\n",
       "      <td>6.7663</td>\n",
       "      <td>1.6420</td>\n",
       "      <td>5.3038</td>\n",
       "    </tr>\n",
       "  </tbody>\n",
       "</table>\n",
       "<p>5 rows × 19190 columns</p>\n",
       "</div>"
      ],
      "text/plain": [
       "            sample   MT-ND1   MT-ND2   MT-CO1   MT-CO2  MT-ATP8  MT-ATP6  \\\n",
       "0  TCGA-02-0047-01  12.9202  13.1220  14.0136  13.5251  13.8439  13.7777   \n",
       "1  TCGA-02-0055-01  10.3972  10.7774  12.3945  12.3893  11.7280  11.9408   \n",
       "2  TCGA-02-2483-01  11.6941  12.0249  13.1153  13.4086  13.6287  13.4227   \n",
       "3  TCGA-02-2485-01  10.4295  10.5171  11.9787  11.6139  12.0474  11.7723   \n",
       "4  TCGA-04-1331-01  14.0694  14.6290  14.2815  14.2857  13.8989  14.3392   \n",
       "\n",
       "    MT-CO3   MT-ND3  MT-ND4L  ...  S100A1   CHTOP  SNAPIN  SH3D21   STK40  \\\n",
       "0  13.4037  12.8123  13.4187  ...  6.4741  5.8017  6.4975  2.1509  5.2445   \n",
       "1  12.2022  11.0142  11.0221  ...  5.9969  5.6888  6.0633  1.9822  4.3299   \n",
       "2  13.4436  12.9984  11.9406  ...  4.8156  6.3632  6.6416  0.0718  4.6445   \n",
       "3  11.7649  10.9137  10.9300  ...  5.0968  6.2915  6.5590  0.2522  4.5009   \n",
       "4  14.2848  13.6645  13.5692  ...  8.8997  6.0978  5.5512  1.7660  4.3198   \n",
       "\n",
       "    LSM10   OSCP1  MRPS15   CSF3R   EVA1B  \n",
       "0  6.0507  3.8915  5.5389  4.2936  4.6685  \n",
       "1  6.4992  3.7582  6.7092  4.5964  5.5145  \n",
       "2  6.0581  3.0515  6.2878  3.5035  3.9856  \n",
       "3  6.0800  3.7367  6.2054  2.6067  3.7039  \n",
       "4  5.1273  3.8360  6.7663  1.6420  5.3038  \n",
       "\n",
       "[5 rows x 19190 columns]"
      ]
     },
     "execution_count": 18,
     "metadata": {},
     "output_type": "execute_result"
    }
   ],
   "source": [
    "# %%time\n",
    "# # DL data gene expression for Normal Tissue and Primary Tumors\n",
    "# path = output_file_directory + \"/ATG7/dataframes_for_input/\"\n",
    "# symbol_normal_protcoding = pandas.read_csv(path+\"symbol_normal_protcoding_expr.tsv\", sep = \"\\t\")\n",
    "# symbol_primary_protcoding = pandas.read_csv(path+\"symbol_primary_protcoding_expr.tsv\", sep = \"\\t\")\n",
    "\n",
    "# print(symbol_normal_protcoding.shape)\n",
    "# print(symbol_primary_protcoding.shape)\n",
    "\n",
    "# symbol_primary_protcoding.head()"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 19,
   "metadata": {},
   "outputs": [
    {
     "name": "stdout",
     "output_type": "stream",
     "text": [
      "(7429, 19194)\n",
      "(9185, 19194)\n"
     ]
    },
    {
     "data": {
      "text/html": [
       "<div>\n",
       "<style scoped>\n",
       "    .dataframe tbody tr th:only-of-type {\n",
       "        vertical-align: middle;\n",
       "    }\n",
       "\n",
       "    .dataframe tbody tr th {\n",
       "        vertical-align: top;\n",
       "    }\n",
       "\n",
       "    .dataframe thead th {\n",
       "        text-align: right;\n",
       "    }\n",
       "</style>\n",
       "<table border=\"1\" class=\"dataframe\">\n",
       "  <thead>\n",
       "    <tr style=\"text-align: right;\">\n",
       "      <th></th>\n",
       "      <th>Sample_Type</th>\n",
       "      <th>Primary_Site</th>\n",
       "      <th>ATG7_x</th>\n",
       "      <th>ATG7_1</th>\n",
       "      <th>ATG7_2</th>\n",
       "      <th>MT-ND1</th>\n",
       "      <th>MT-ND2</th>\n",
       "      <th>MT-CO1</th>\n",
       "      <th>MT-CO2</th>\n",
       "      <th>MT-ATP8</th>\n",
       "      <th>...</th>\n",
       "      <th>S100A1</th>\n",
       "      <th>CHTOP</th>\n",
       "      <th>SNAPIN</th>\n",
       "      <th>SH3D21</th>\n",
       "      <th>STK40</th>\n",
       "      <th>LSM10</th>\n",
       "      <th>OSCP1</th>\n",
       "      <th>MRPS15</th>\n",
       "      <th>CSF3R</th>\n",
       "      <th>EVA1B</th>\n",
       "    </tr>\n",
       "  </thead>\n",
       "  <tbody>\n",
       "    <tr>\n",
       "      <th>0</th>\n",
       "      <td>Normal Tissue</td>\n",
       "      <td>Muscle</td>\n",
       "      <td>5.042</td>\n",
       "      <td>3.860</td>\n",
       "      <td>2.390</td>\n",
       "      <td>14.0639</td>\n",
       "      <td>13.9595</td>\n",
       "      <td>13.8503</td>\n",
       "      <td>14.4453</td>\n",
       "      <td>15.0971</td>\n",
       "      <td>...</td>\n",
       "      <td>2.7868</td>\n",
       "      <td>5.6494</td>\n",
       "      <td>5.0193</td>\n",
       "      <td>2.6940</td>\n",
       "      <td>4.5274</td>\n",
       "      <td>5.0998</td>\n",
       "      <td>2.3423</td>\n",
       "      <td>6.0790</td>\n",
       "      <td>6.4861</td>\n",
       "      <td>4.9700</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>1</th>\n",
       "      <td>Normal Tissue</td>\n",
       "      <td>Muscle</td>\n",
       "      <td>4.252</td>\n",
       "      <td>2.322</td>\n",
       "      <td>1.956</td>\n",
       "      <td>13.9265</td>\n",
       "      <td>13.5455</td>\n",
       "      <td>15.2642</td>\n",
       "      <td>14.3855</td>\n",
       "      <td>15.1867</td>\n",
       "      <td>...</td>\n",
       "      <td>9.3501</td>\n",
       "      <td>6.1104</td>\n",
       "      <td>6.0570</td>\n",
       "      <td>1.1770</td>\n",
       "      <td>6.8792</td>\n",
       "      <td>6.4325</td>\n",
       "      <td>2.1013</td>\n",
       "      <td>6.7035</td>\n",
       "      <td>1.1706</td>\n",
       "      <td>1.3167</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>2</th>\n",
       "      <td>Normal Tissue</td>\n",
       "      <td>Muscle</td>\n",
       "      <td>3.909</td>\n",
       "      <td>2.602</td>\n",
       "      <td>1.345</td>\n",
       "      <td>14.2878</td>\n",
       "      <td>14.1511</td>\n",
       "      <td>14.3722</td>\n",
       "      <td>13.8211</td>\n",
       "      <td>14.8161</td>\n",
       "      <td>...</td>\n",
       "      <td>9.0646</td>\n",
       "      <td>5.9750</td>\n",
       "      <td>5.1173</td>\n",
       "      <td>1.8683</td>\n",
       "      <td>4.6136</td>\n",
       "      <td>5.0090</td>\n",
       "      <td>2.0604</td>\n",
       "      <td>5.7848</td>\n",
       "      <td>1.7489</td>\n",
       "      <td>4.6514</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>3</th>\n",
       "      <td>Normal Tissue</td>\n",
       "      <td>Muscle</td>\n",
       "      <td>3.854</td>\n",
       "      <td>2.568</td>\n",
       "      <td>1.428</td>\n",
       "      <td>13.8853</td>\n",
       "      <td>14.2716</td>\n",
       "      <td>15.1167</td>\n",
       "      <td>14.0622</td>\n",
       "      <td>15.5897</td>\n",
       "      <td>...</td>\n",
       "      <td>9.2252</td>\n",
       "      <td>5.4941</td>\n",
       "      <td>5.5880</td>\n",
       "      <td>-0.4521</td>\n",
       "      <td>5.6631</td>\n",
       "      <td>6.3797</td>\n",
       "      <td>2.5683</td>\n",
       "      <td>7.3214</td>\n",
       "      <td>1.6140</td>\n",
       "      <td>1.5563</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>4</th>\n",
       "      <td>Normal Tissue</td>\n",
       "      <td>Muscle</td>\n",
       "      <td>3.822</td>\n",
       "      <td>2.164</td>\n",
       "      <td>1.293</td>\n",
       "      <td>12.9986</td>\n",
       "      <td>13.7570</td>\n",
       "      <td>14.4661</td>\n",
       "      <td>13.4082</td>\n",
       "      <td>14.0564</td>\n",
       "      <td>...</td>\n",
       "      <td>8.9779</td>\n",
       "      <td>6.1502</td>\n",
       "      <td>5.1330</td>\n",
       "      <td>2.0289</td>\n",
       "      <td>6.8202</td>\n",
       "      <td>6.1789</td>\n",
       "      <td>1.8524</td>\n",
       "      <td>6.7685</td>\n",
       "      <td>2.8838</td>\n",
       "      <td>1.8323</td>\n",
       "    </tr>\n",
       "  </tbody>\n",
       "</table>\n",
       "<p>5 rows × 19194 columns</p>\n",
       "</div>"
      ],
      "text/plain": [
       "     Sample_Type Primary_Site  ATG7_x  ATG7_1  ATG7_2   MT-ND1   MT-ND2  \\\n",
       "0  Normal Tissue       Muscle   5.042   3.860   2.390  14.0639  13.9595   \n",
       "1  Normal Tissue       Muscle   4.252   2.322   1.956  13.9265  13.5455   \n",
       "2  Normal Tissue       Muscle   3.909   2.602   1.345  14.2878  14.1511   \n",
       "3  Normal Tissue       Muscle   3.854   2.568   1.428  13.8853  14.2716   \n",
       "4  Normal Tissue       Muscle   3.822   2.164   1.293  12.9986  13.7570   \n",
       "\n",
       "    MT-CO1   MT-CO2  MT-ATP8  ...  S100A1   CHTOP  SNAPIN  SH3D21   STK40  \\\n",
       "0  13.8503  14.4453  15.0971  ...  2.7868  5.6494  5.0193  2.6940  4.5274   \n",
       "1  15.2642  14.3855  15.1867  ...  9.3501  6.1104  6.0570  1.1770  6.8792   \n",
       "2  14.3722  13.8211  14.8161  ...  9.0646  5.9750  5.1173  1.8683  4.6136   \n",
       "3  15.1167  14.0622  15.5897  ...  9.2252  5.4941  5.5880 -0.4521  5.6631   \n",
       "4  14.4661  13.4082  14.0564  ...  8.9779  6.1502  5.1330  2.0289  6.8202   \n",
       "\n",
       "    LSM10   OSCP1  MRPS15   CSF3R   EVA1B  \n",
       "0  5.0998  2.3423  6.0790  6.4861  4.9700  \n",
       "1  6.4325  2.1013  6.7035  1.1706  1.3167  \n",
       "2  5.0090  2.0604  5.7848  1.7489  4.6514  \n",
       "3  6.3797  2.5683  7.3214  1.6140  1.5563  \n",
       "4  6.1789  1.8524  6.7685  2.8838  1.8323  \n",
       "\n",
       "[5 rows x 19194 columns]"
      ]
     },
     "execution_count": 19,
     "metadata": {},
     "output_type": "execute_result"
    }
   ],
   "source": [
    "# #Preparation of the global dataframe with ATG7 isoform expression AND whole gene expression\n",
    "# data = [[symbol_normal_protcoding, df_ATG7_Normal], [symbol_primary_protcoding, df_ATG7_Primary]]\n",
    "\n",
    "# df_final = []\n",
    "\n",
    "# for df in data:\n",
    "#     #Merge the two df to have ATG7 isoform expression + whole gene expression \n",
    "#   df_merged = pandas.merge(df[1], df[0], on= 'sample')\n",
    "\n",
    "#   #delete the sample column and select column\n",
    "#   df_merged = df_merged.drop(\"sample\", 1)\n",
    "#   df_merged = df_merged.drop(df_merged.columns[5:19], axis=1)\n",
    "  \n",
    "#   #save in list\n",
    "#   df_final.append(df_merged)\n",
    "\n",
    "# symbol_normal_protcoding_atg7_expr = df_final[0]\n",
    "# symbol_primary_protcoding_atg7_expr = df_final[1]\n",
    "\n",
    "# print(symbol_normal_protcoding_atg7_expr.shape)\n",
    "# print(symbol_primary_protcoding_atg7_expr.shape)\n",
    "# symbol_normal_protcoding_atg7_expr.head()"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 20,
   "metadata": {},
   "outputs": [
    {
     "name": "stdout",
     "output_type": "stream",
     "text": [
      "   ATG7_x  ATG7_1  ATG7_2  ATG7_y\n",
      "0   5.042   3.860   2.390  5.0418\n",
      "1   4.252   2.322   1.956  4.2518\n"
     ]
    }
   ],
   "source": [
    "# #Why ATG7_x ?\n",
    "# verif = symbol_normal_protcoding_atg7_expr[symbol_normal_protcoding_atg7_expr.columns[pandas.Series(symbol_normal_protcoding_atg7_expr.columns).str.startswith('ATG7')]]\n",
    "# print(verif.head(2))"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 21,
   "metadata": {},
   "outputs": [
    {
     "name": "stdout",
     "output_type": "stream",
     "text": [
      "    ATG7  ATG7_1  ATG7_2\n",
      "0  5.042   3.860   2.390\n",
      "1  4.252   2.322   1.956\n",
      "    ATG7  ATG7_1  ATG7_2\n",
      "0  4.824   3.239   1.903\n",
      "1  4.790   4.056   1.151\n"
     ]
    }
   ],
   "source": [
    "# data = [symbol_normal_protcoding_atg7_expr, symbol_primary_protcoding_atg7_expr]\n",
    "\n",
    "# for df in data:\n",
    "#     df.drop('ATG7_y', axis = 1, inplace=True) \n",
    "#     df.rename(columns = {'ATG7_x' : 'ATG7'}, inplace=True)\n",
    "#     verif = df[df.columns[pandas.Series(df.columns).str.startswith('ATG7')]]\n",
    "#     print(verif.head(2))"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 22,
   "metadata": {},
   "outputs": [],
   "source": [
    "# #save to csv\n",
    "\n",
    "# symbol_normal_protcoding_atg7_expr.to_csv(path+path2+'symbol_normal_atg7_ProteinGE.tsv',sep = \"\\t\", index=False)\n",
    "# symbol_primary_protcoding_atg7_expr.to_csv(path+path2+'symbol_primary_atg7_ProteinGE.tsv',sep = \"\\t\", index=False)"
   ]
  }
 ],
 "metadata": {
  "colab": {
   "collapsed_sections": [
    "scwtwuDCqUKz",
    "fbjL51C8RltI",
    "R8fLWHXOBYJl"
   ],
   "name": "all_organs.v004.ipynb",
   "provenance": []
  },
  "kernelspec": {
   "display_name": "Python 3",
   "language": "python",
   "name": "python3"
  },
  "language_info": {
   "codemirror_mode": {
    "name": "ipython",
    "version": 3
   },
   "file_extension": ".py",
   "mimetype": "text/x-python",
   "name": "python",
   "nbconvert_exporter": "python",
   "pygments_lexer": "ipython3",
   "version": "3.8.5"
  }
 },
 "nbformat": 4,
 "nbformat_minor": 1
}
