{
 "cells": [
  {
   "cell_type": "markdown",
   "metadata": {
    "id": "ifhgKXirU9Ds"
   },
   "source": [
    "# I] Import and download"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 16,
   "metadata": {
    "id": "TLQwwG5VDPFA"
   },
   "outputs": [],
   "source": [
    "import pandas"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 17,
   "metadata": {},
   "outputs": [],
   "source": [
    "input_file_directory = '/Users/kja11/OneDrive - Háskóli Íslands/PhD ATG7//0 in_silico/Python/1)data_input/'"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 18,
   "metadata": {},
   "outputs": [],
   "source": [
    "output_file_directory = '/Users/kja11/OneDrive - Háskóli Íslands/PhD ATG7/0 in_silico/Python/3)output/'"
   ]
  },
  {
   "cell_type": "markdown",
   "metadata": {},
   "source": [
    "## A) GABARAP isoform expression "
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 19,
   "metadata": {},
   "outputs": [
    {
     "name": "stdout",
     "output_type": "stream",
     "text": [
      "(16614, 20)\n",
      "Wall time: 71.4 ms\n"
     ]
    },
    {
     "data": {
      "text/html": [
       "<div>\n",
       "<style scoped>\n",
       "    .dataframe tbody tr th:only-of-type {\n",
       "        vertical-align: middle;\n",
       "    }\n",
       "\n",
       "    .dataframe tbody tr th {\n",
       "        vertical-align: top;\n",
       "    }\n",
       "\n",
       "    .dataframe thead th {\n",
       "        text-align: right;\n",
       "    }\n",
       "</style>\n",
       "<table border=\"1\" class=\"dataframe\">\n",
       "  <thead>\n",
       "    <tr style=\"text-align: right;\">\n",
       "      <th></th>\n",
       "      <th>sample</th>\n",
       "      <th>Sample_Type</th>\n",
       "      <th>Primary_Site</th>\n",
       "      <th>GABARAP</th>\n",
       "      <th>GABARAP_1</th>\n",
       "      <th>GABARAP_2</th>\n",
       "      <th>GABARAP_3</th>\n",
       "      <th>GABARAP_total</th>\n",
       "      <th>GABARAP_1tpm</th>\n",
       "      <th>GABARAP_2tpm</th>\n",
       "      <th>GABARAP_3tpm</th>\n",
       "      <th>log2_p1_GABARAP_total</th>\n",
       "      <th>log2_p1_GABARAP_1</th>\n",
       "      <th>log2_p1_GABARAP_2</th>\n",
       "      <th>GABARAP_prot_tot</th>\n",
       "      <th>GABARAP_1perc</th>\n",
       "      <th>GABARAP_2perc</th>\n",
       "      <th>GABARAP_3perc</th>\n",
       "      <th>GABARAP_1norm</th>\n",
       "      <th>GABARAP_2norm</th>\n",
       "    </tr>\n",
       "  </thead>\n",
       "  <tbody>\n",
       "    <tr>\n",
       "      <th>0</th>\n",
       "      <td>GTEX-OIZI-0005-SM-2XCED</td>\n",
       "      <td>Normal Tissue</td>\n",
       "      <td>Blood</td>\n",
       "      <td>11.27</td>\n",
       "      <td>7.823</td>\n",
       "      <td>9.316</td>\n",
       "      <td>10.40</td>\n",
       "      <td>2469.493431</td>\n",
       "      <td>226.441351</td>\n",
       "      <td>637.374609</td>\n",
       "      <td>1351.175101</td>\n",
       "      <td>11.270584</td>\n",
       "      <td>7.829351</td>\n",
       "      <td>9.318259</td>\n",
       "      <td>2214.991061</td>\n",
       "      <td>10.223127</td>\n",
       "      <td>28.775493</td>\n",
       "      <td>61.001379</td>\n",
       "      <td>0.091695</td>\n",
       "      <td>0.258099</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>1</th>\n",
       "      <td>GTEX-XAJ8-0006-SM-46MVM</td>\n",
       "      <td>Normal Tissue</td>\n",
       "      <td>Blood</td>\n",
       "      <td>11.24</td>\n",
       "      <td>6.587</td>\n",
       "      <td>9.797</td>\n",
       "      <td>10.40</td>\n",
       "      <td>2418.671971</td>\n",
       "      <td>96.134675</td>\n",
       "      <td>889.590998</td>\n",
       "      <td>1351.175101</td>\n",
       "      <td>11.240596</td>\n",
       "      <td>6.601914</td>\n",
       "      <td>9.798619</td>\n",
       "      <td>2336.900774</td>\n",
       "      <td>4.113768</td>\n",
       "      <td>38.067128</td>\n",
       "      <td>57.819105</td>\n",
       "      <td>0.039747</td>\n",
       "      <td>0.367801</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>2</th>\n",
       "      <td>GTEX-ZP4G-0006-SM-4WWE6</td>\n",
       "      <td>Normal Tissue</td>\n",
       "      <td>Blood</td>\n",
       "      <td>11.18</td>\n",
       "      <td>6.138</td>\n",
       "      <td>9.749</td>\n",
       "      <td>10.36</td>\n",
       "      <td>2320.145197</td>\n",
       "      <td>70.423231</td>\n",
       "      <td>860.480282</td>\n",
       "      <td>1314.227119</td>\n",
       "      <td>11.180621</td>\n",
       "      <td>6.158321</td>\n",
       "      <td>9.750674</td>\n",
       "      <td>2245.130632</td>\n",
       "      <td>3.136710</td>\n",
       "      <td>38.326513</td>\n",
       "      <td>58.536777</td>\n",
       "      <td>0.030353</td>\n",
       "      <td>0.370873</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>3</th>\n",
       "      <td>GTEX-WFG8-0006-SM-3GIKS</td>\n",
       "      <td>Normal Tissue</td>\n",
       "      <td>Blood</td>\n",
       "      <td>11.17</td>\n",
       "      <td>6.901</td>\n",
       "      <td>9.374</td>\n",
       "      <td>10.49</td>\n",
       "      <td>2304.118777</td>\n",
       "      <td>119.510033</td>\n",
       "      <td>663.520774</td>\n",
       "      <td>1438.150553</td>\n",
       "      <td>11.170625</td>\n",
       "      <td>6.913009</td>\n",
       "      <td>9.376170</td>\n",
       "      <td>2221.181360</td>\n",
       "      <td>5.380472</td>\n",
       "      <td>29.872427</td>\n",
       "      <td>64.747102</td>\n",
       "      <td>0.051868</td>\n",
       "      <td>0.287972</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>4</th>\n",
       "      <td>GTEX-QCQG-0006-SM-5SI8M</td>\n",
       "      <td>Normal Tissue</td>\n",
       "      <td>Blood</td>\n",
       "      <td>11.08</td>\n",
       "      <td>6.793</td>\n",
       "      <td>9.684</td>\n",
       "      <td>10.15</td>\n",
       "      <td>2164.771947</td>\n",
       "      <td>110.890118</td>\n",
       "      <td>822.572053</td>\n",
       "      <td>1136.198139</td>\n",
       "      <td>11.080666</td>\n",
       "      <td>6.805939</td>\n",
       "      <td>9.685751</td>\n",
       "      <td>2069.660310</td>\n",
       "      <td>5.357890</td>\n",
       "      <td>39.744302</td>\n",
       "      <td>54.897808</td>\n",
       "      <td>0.051225</td>\n",
       "      <td>0.379981</td>\n",
       "    </tr>\n",
       "  </tbody>\n",
       "</table>\n",
       "</div>"
      ],
      "text/plain": [
       "                    sample    Sample_Type Primary_Site  GABARAP  GABARAP_1  \\\n",
       "0  GTEX-OIZI-0005-SM-2XCED  Normal Tissue        Blood    11.27      7.823   \n",
       "1  GTEX-XAJ8-0006-SM-46MVM  Normal Tissue        Blood    11.24      6.587   \n",
       "2  GTEX-ZP4G-0006-SM-4WWE6  Normal Tissue        Blood    11.18      6.138   \n",
       "3  GTEX-WFG8-0006-SM-3GIKS  Normal Tissue        Blood    11.17      6.901   \n",
       "4  GTEX-QCQG-0006-SM-5SI8M  Normal Tissue        Blood    11.08      6.793   \n",
       "\n",
       "   GABARAP_2  GABARAP_3  GABARAP_total  GABARAP_1tpm  GABARAP_2tpm  \\\n",
       "0      9.316      10.40    2469.493431    226.441351    637.374609   \n",
       "1      9.797      10.40    2418.671971     96.134675    889.590998   \n",
       "2      9.749      10.36    2320.145197     70.423231    860.480282   \n",
       "3      9.374      10.49    2304.118777    119.510033    663.520774   \n",
       "4      9.684      10.15    2164.771947    110.890118    822.572053   \n",
       "\n",
       "   GABARAP_3tpm  log2_p1_GABARAP_total  log2_p1_GABARAP_1  log2_p1_GABARAP_2  \\\n",
       "0   1351.175101              11.270584           7.829351           9.318259   \n",
       "1   1351.175101              11.240596           6.601914           9.798619   \n",
       "2   1314.227119              11.180621           6.158321           9.750674   \n",
       "3   1438.150553              11.170625           6.913009           9.376170   \n",
       "4   1136.198139              11.080666           6.805939           9.685751   \n",
       "\n",
       "   GABARAP_prot_tot  GABARAP_1perc  GABARAP_2perc  GABARAP_3perc  \\\n",
       "0       2214.991061      10.223127      28.775493      61.001379   \n",
       "1       2336.900774       4.113768      38.067128      57.819105   \n",
       "2       2245.130632       3.136710      38.326513      58.536777   \n",
       "3       2221.181360       5.380472      29.872427      64.747102   \n",
       "4       2069.660310       5.357890      39.744302      54.897808   \n",
       "\n",
       "   GABARAP_1norm  GABARAP_2norm  \n",
       "0       0.091695       0.258099  \n",
       "1       0.039747       0.367801  \n",
       "2       0.030353       0.370873  \n",
       "3       0.051868       0.287972  \n",
       "4       0.051225       0.379981  "
      ]
     },
     "execution_count": 19,
     "metadata": {},
     "output_type": "execute_result"
    }
   ],
   "source": [
    "%%time\n",
    "#DL data GABARAP isoform expression\n",
    "path = output_file_directory + \"/GABARAP/dataframes_for_input/df_GABARAP_NormPrim.csv\"\n",
    "df_GABARAP_NormPrim= pandas.read_csv(path, sep = \",\")\n",
    "\n",
    "print(df_GABARAP_NormPrim.shape)\n",
    "df_GABARAP_NormPrim.head()"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 20,
   "metadata": {},
   "outputs": [
    {
     "name": "stdout",
     "output_type": "stream",
     "text": [
      "7429\n",
      "9185\n"
     ]
    }
   ],
   "source": [
    "#split to have Normal (GTEX) and Primary (TCGA)\n",
    "df_GABARAP_Normal = df_GABARAP_NormPrim[df_GABARAP_NormPrim['sample'].str.startswith('GTEX')]\n",
    "print(len(df_GABARAP_Normal))\n",
    "df_GABARAP_Primary = df_GABARAP_NormPrim[df_GABARAP_NormPrim['sample'].str.startswith('TCGA')]\n",
    "print(len(df_GABARAP_Primary))"
   ]
  },
  {
   "cell_type": "markdown",
   "metadata": {},
   "source": [
    "## B) GABARAP all gene expression"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 21,
   "metadata": {},
   "outputs": [
    {
     "name": "stdout",
     "output_type": "stream",
     "text": [
      "(7429, 19383)\n",
      "(9185, 19383)\n",
      "Wall time: 3min 32s\n"
     ]
    },
    {
     "data": {
      "text/html": [
       "<div>\n",
       "<style scoped>\n",
       "    .dataframe tbody tr th:only-of-type {\n",
       "        vertical-align: middle;\n",
       "    }\n",
       "\n",
       "    .dataframe tbody tr th {\n",
       "        vertical-align: top;\n",
       "    }\n",
       "\n",
       "    .dataframe thead th {\n",
       "        text-align: right;\n",
       "    }\n",
       "</style>\n",
       "<table border=\"1\" class=\"dataframe\">\n",
       "  <thead>\n",
       "    <tr style=\"text-align: right;\">\n",
       "      <th></th>\n",
       "      <th>sample</th>\n",
       "      <th>ENSG00000198888</th>\n",
       "      <th>ENSG00000198763</th>\n",
       "      <th>ENSG00000198804</th>\n",
       "      <th>ENSG00000198712</th>\n",
       "      <th>ENSG00000228253</th>\n",
       "      <th>ENSG00000198899</th>\n",
       "      <th>ENSG00000198938</th>\n",
       "      <th>ENSG00000198840</th>\n",
       "      <th>ENSG00000212907</th>\n",
       "      <th>...</th>\n",
       "      <th>ENSG00000160678</th>\n",
       "      <th>ENSG00000160679</th>\n",
       "      <th>ENSG00000143553</th>\n",
       "      <th>ENSG00000214193</th>\n",
       "      <th>ENSG00000196182</th>\n",
       "      <th>ENSG00000181817</th>\n",
       "      <th>ENSG00000116885</th>\n",
       "      <th>ENSG00000116898</th>\n",
       "      <th>ENSG00000119535</th>\n",
       "      <th>ENSG00000142694</th>\n",
       "    </tr>\n",
       "  </thead>\n",
       "  <tbody>\n",
       "    <tr>\n",
       "      <th>0</th>\n",
       "      <td>TCGA-02-0047-01</td>\n",
       "      <td>12.9202</td>\n",
       "      <td>13.1220</td>\n",
       "      <td>14.0136</td>\n",
       "      <td>13.5251</td>\n",
       "      <td>13.8439</td>\n",
       "      <td>13.7777</td>\n",
       "      <td>13.4037</td>\n",
       "      <td>12.8123</td>\n",
       "      <td>13.4187</td>\n",
       "      <td>...</td>\n",
       "      <td>6.4741</td>\n",
       "      <td>5.8017</td>\n",
       "      <td>6.4975</td>\n",
       "      <td>2.1509</td>\n",
       "      <td>5.2445</td>\n",
       "      <td>6.0507</td>\n",
       "      <td>3.8915</td>\n",
       "      <td>5.5389</td>\n",
       "      <td>4.2936</td>\n",
       "      <td>4.6685</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>1</th>\n",
       "      <td>TCGA-02-0055-01</td>\n",
       "      <td>10.3972</td>\n",
       "      <td>10.7774</td>\n",
       "      <td>12.3945</td>\n",
       "      <td>12.3893</td>\n",
       "      <td>11.7280</td>\n",
       "      <td>11.9408</td>\n",
       "      <td>12.2022</td>\n",
       "      <td>11.0142</td>\n",
       "      <td>11.0221</td>\n",
       "      <td>...</td>\n",
       "      <td>5.9969</td>\n",
       "      <td>5.6888</td>\n",
       "      <td>6.0633</td>\n",
       "      <td>1.9822</td>\n",
       "      <td>4.3299</td>\n",
       "      <td>6.4992</td>\n",
       "      <td>3.7582</td>\n",
       "      <td>6.7092</td>\n",
       "      <td>4.5964</td>\n",
       "      <td>5.5145</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>2</th>\n",
       "      <td>TCGA-02-2483-01</td>\n",
       "      <td>11.6941</td>\n",
       "      <td>12.0249</td>\n",
       "      <td>13.1153</td>\n",
       "      <td>13.4086</td>\n",
       "      <td>13.6287</td>\n",
       "      <td>13.4227</td>\n",
       "      <td>13.4436</td>\n",
       "      <td>12.9984</td>\n",
       "      <td>11.9406</td>\n",
       "      <td>...</td>\n",
       "      <td>4.8156</td>\n",
       "      <td>6.3632</td>\n",
       "      <td>6.6416</td>\n",
       "      <td>0.0718</td>\n",
       "      <td>4.6445</td>\n",
       "      <td>6.0581</td>\n",
       "      <td>3.0515</td>\n",
       "      <td>6.2878</td>\n",
       "      <td>3.5035</td>\n",
       "      <td>3.9856</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>3</th>\n",
       "      <td>TCGA-02-2485-01</td>\n",
       "      <td>10.4295</td>\n",
       "      <td>10.5171</td>\n",
       "      <td>11.9787</td>\n",
       "      <td>11.6139</td>\n",
       "      <td>12.0474</td>\n",
       "      <td>11.7723</td>\n",
       "      <td>11.7649</td>\n",
       "      <td>10.9137</td>\n",
       "      <td>10.9300</td>\n",
       "      <td>...</td>\n",
       "      <td>5.0968</td>\n",
       "      <td>6.2915</td>\n",
       "      <td>6.5590</td>\n",
       "      <td>0.2522</td>\n",
       "      <td>4.5009</td>\n",
       "      <td>6.0800</td>\n",
       "      <td>3.7367</td>\n",
       "      <td>6.2054</td>\n",
       "      <td>2.6067</td>\n",
       "      <td>3.7039</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>4</th>\n",
       "      <td>TCGA-04-1331-01</td>\n",
       "      <td>14.0694</td>\n",
       "      <td>14.6290</td>\n",
       "      <td>14.2815</td>\n",
       "      <td>14.2857</td>\n",
       "      <td>13.8989</td>\n",
       "      <td>14.3392</td>\n",
       "      <td>14.2848</td>\n",
       "      <td>13.6645</td>\n",
       "      <td>13.5692</td>\n",
       "      <td>...</td>\n",
       "      <td>8.8997</td>\n",
       "      <td>6.0978</td>\n",
       "      <td>5.5512</td>\n",
       "      <td>1.7660</td>\n",
       "      <td>4.3198</td>\n",
       "      <td>5.1273</td>\n",
       "      <td>3.8360</td>\n",
       "      <td>6.7663</td>\n",
       "      <td>1.6420</td>\n",
       "      <td>5.3038</td>\n",
       "    </tr>\n",
       "  </tbody>\n",
       "</table>\n",
       "<p>5 rows × 19383 columns</p>\n",
       "</div>"
      ],
      "text/plain": [
       "            sample  ENSG00000198888  ENSG00000198763  ENSG00000198804  \\\n",
       "0  TCGA-02-0047-01          12.9202          13.1220          14.0136   \n",
       "1  TCGA-02-0055-01          10.3972          10.7774          12.3945   \n",
       "2  TCGA-02-2483-01          11.6941          12.0249          13.1153   \n",
       "3  TCGA-02-2485-01          10.4295          10.5171          11.9787   \n",
       "4  TCGA-04-1331-01          14.0694          14.6290          14.2815   \n",
       "\n",
       "   ENSG00000198712  ENSG00000228253  ENSG00000198899  ENSG00000198938  \\\n",
       "0          13.5251          13.8439          13.7777          13.4037   \n",
       "1          12.3893          11.7280          11.9408          12.2022   \n",
       "2          13.4086          13.6287          13.4227          13.4436   \n",
       "3          11.6139          12.0474          11.7723          11.7649   \n",
       "4          14.2857          13.8989          14.3392          14.2848   \n",
       "\n",
       "   ENSG00000198840  ENSG00000212907  ...  ENSG00000160678  ENSG00000160679  \\\n",
       "0          12.8123          13.4187  ...           6.4741           5.8017   \n",
       "1          11.0142          11.0221  ...           5.9969           5.6888   \n",
       "2          12.9984          11.9406  ...           4.8156           6.3632   \n",
       "3          10.9137          10.9300  ...           5.0968           6.2915   \n",
       "4          13.6645          13.5692  ...           8.8997           6.0978   \n",
       "\n",
       "   ENSG00000143553  ENSG00000214193  ENSG00000196182  ENSG00000181817  \\\n",
       "0           6.4975           2.1509           5.2445           6.0507   \n",
       "1           6.0633           1.9822           4.3299           6.4992   \n",
       "2           6.6416           0.0718           4.6445           6.0581   \n",
       "3           6.5590           0.2522           4.5009           6.0800   \n",
       "4           5.5512           1.7660           4.3198           5.1273   \n",
       "\n",
       "   ENSG00000116885  ENSG00000116898  ENSG00000119535  ENSG00000142694  \n",
       "0           3.8915           5.5389           4.2936           4.6685  \n",
       "1           3.7582           6.7092           4.5964           5.5145  \n",
       "2           3.0515           6.2878           3.5035           3.9856  \n",
       "3           3.7367           6.2054           2.6067           3.7039  \n",
       "4           3.8360           6.7663           1.6420           5.3038  \n",
       "\n",
       "[5 rows x 19383 columns]"
      ]
     },
     "execution_count": 21,
     "metadata": {},
     "output_type": "execute_result"
    }
   ],
   "source": [
    "%%time\n",
    "# DL data gene expression for Normal Tissue and Primary Tumors\n",
    "path = output_file_directory + \"dataframes_for_input/\"\n",
    "ensembl_normal_protcoding = pandas.read_csv(path+\"ensembl_normal_protcoding_expr.tsv\", sep = \"\\t\")\n",
    "ensembl_primary_protcoding = pandas.read_csv(path+\"ensembl_primary_protcoding_expr.tsv\", sep = \"\\t\")\n",
    "\n",
    "print(ensembl_normal_protcoding.shape)\n",
    "print(ensembl_primary_protcoding.shape)\n",
    "\n",
    "ensembl_primary_protcoding.head()"
   ]
  },
  {
   "cell_type": "markdown",
   "metadata": {},
   "source": [
    "# II] GABARAP isoforms + Whole gene expression (ensembl)"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 22,
   "metadata": {
    "id": "9b0Bg43gEMRM",
    "scrolled": true
   },
   "outputs": [
    {
     "name": "stdout",
     "output_type": "stream",
     "text": [
      "(7429, 19387)\n",
      "(9185, 19387)\n"
     ]
    },
    {
     "data": {
      "text/html": [
       "<div>\n",
       "<style scoped>\n",
       "    .dataframe tbody tr th:only-of-type {\n",
       "        vertical-align: middle;\n",
       "    }\n",
       "\n",
       "    .dataframe tbody tr th {\n",
       "        vertical-align: top;\n",
       "    }\n",
       "\n",
       "    .dataframe thead th {\n",
       "        text-align: right;\n",
       "    }\n",
       "</style>\n",
       "<table border=\"1\" class=\"dataframe\">\n",
       "  <thead>\n",
       "    <tr style=\"text-align: right;\">\n",
       "      <th></th>\n",
       "      <th>Sample_Type</th>\n",
       "      <th>Primary_Site</th>\n",
       "      <th>GABARAP</th>\n",
       "      <th>GABARAP_1</th>\n",
       "      <th>GABARAP_2</th>\n",
       "      <th>ENSG00000198888</th>\n",
       "      <th>ENSG00000198763</th>\n",
       "      <th>ENSG00000198804</th>\n",
       "      <th>ENSG00000198712</th>\n",
       "      <th>ENSG00000228253</th>\n",
       "      <th>...</th>\n",
       "      <th>ENSG00000160678</th>\n",
       "      <th>ENSG00000160679</th>\n",
       "      <th>ENSG00000143553</th>\n",
       "      <th>ENSG00000214193</th>\n",
       "      <th>ENSG00000196182</th>\n",
       "      <th>ENSG00000181817</th>\n",
       "      <th>ENSG00000116885</th>\n",
       "      <th>ENSG00000116898</th>\n",
       "      <th>ENSG00000119535</th>\n",
       "      <th>ENSG00000142694</th>\n",
       "    </tr>\n",
       "  </thead>\n",
       "  <tbody>\n",
       "    <tr>\n",
       "      <th>0</th>\n",
       "      <td>Normal Tissue</td>\n",
       "      <td>Blood</td>\n",
       "      <td>11.27</td>\n",
       "      <td>7.823</td>\n",
       "      <td>9.316</td>\n",
       "      <td>11.4278</td>\n",
       "      <td>11.9220</td>\n",
       "      <td>12.4100</td>\n",
       "      <td>13.1565</td>\n",
       "      <td>12.9439</td>\n",
       "      <td>...</td>\n",
       "      <td>2.1313</td>\n",
       "      <td>5.8726</td>\n",
       "      <td>4.8314</td>\n",
       "      <td>1.0433</td>\n",
       "      <td>6.1056</td>\n",
       "      <td>6.3116</td>\n",
       "      <td>1.4494</td>\n",
       "      <td>5.0527</td>\n",
       "      <td>11.9031</td>\n",
       "      <td>1.8484</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>1</th>\n",
       "      <td>Normal Tissue</td>\n",
       "      <td>Blood</td>\n",
       "      <td>11.24</td>\n",
       "      <td>6.587</td>\n",
       "      <td>9.797</td>\n",
       "      <td>11.1244</td>\n",
       "      <td>10.6001</td>\n",
       "      <td>11.5536</td>\n",
       "      <td>12.0039</td>\n",
       "      <td>11.6778</td>\n",
       "      <td>...</td>\n",
       "      <td>-1.8314</td>\n",
       "      <td>5.2601</td>\n",
       "      <td>4.2442</td>\n",
       "      <td>0.6050</td>\n",
       "      <td>6.9870</td>\n",
       "      <td>6.0466</td>\n",
       "      <td>-0.5125</td>\n",
       "      <td>4.4894</td>\n",
       "      <td>12.6822</td>\n",
       "      <td>4.4700</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>2</th>\n",
       "      <td>Normal Tissue</td>\n",
       "      <td>Blood</td>\n",
       "      <td>11.18</td>\n",
       "      <td>6.138</td>\n",
       "      <td>9.749</td>\n",
       "      <td>10.9395</td>\n",
       "      <td>10.4427</td>\n",
       "      <td>11.5286</td>\n",
       "      <td>12.1191</td>\n",
       "      <td>11.9713</td>\n",
       "      <td>...</td>\n",
       "      <td>-0.0574</td>\n",
       "      <td>5.7581</td>\n",
       "      <td>5.2419</td>\n",
       "      <td>0.5568</td>\n",
       "      <td>6.2610</td>\n",
       "      <td>6.3896</td>\n",
       "      <td>-0.4921</td>\n",
       "      <td>4.8575</td>\n",
       "      <td>12.4269</td>\n",
       "      <td>5.0553</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>3</th>\n",
       "      <td>Normal Tissue</td>\n",
       "      <td>Blood</td>\n",
       "      <td>11.17</td>\n",
       "      <td>6.901</td>\n",
       "      <td>9.374</td>\n",
       "      <td>10.6507</td>\n",
       "      <td>10.4939</td>\n",
       "      <td>12.1795</td>\n",
       "      <td>11.8817</td>\n",
       "      <td>11.2718</td>\n",
       "      <td>...</td>\n",
       "      <td>0.3907</td>\n",
       "      <td>5.9556</td>\n",
       "      <td>5.6491</td>\n",
       "      <td>-0.8339</td>\n",
       "      <td>6.0381</td>\n",
       "      <td>5.9619</td>\n",
       "      <td>-2.6349</td>\n",
       "      <td>3.6804</td>\n",
       "      <td>11.9587</td>\n",
       "      <td>1.0222</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>4</th>\n",
       "      <td>Normal Tissue</td>\n",
       "      <td>Blood</td>\n",
       "      <td>11.08</td>\n",
       "      <td>6.793</td>\n",
       "      <td>9.684</td>\n",
       "      <td>11.3773</td>\n",
       "      <td>10.9368</td>\n",
       "      <td>11.5022</td>\n",
       "      <td>12.1524</td>\n",
       "      <td>12.4101</td>\n",
       "      <td>...</td>\n",
       "      <td>-0.0877</td>\n",
       "      <td>5.8043</td>\n",
       "      <td>3.8973</td>\n",
       "      <td>0.3573</td>\n",
       "      <td>7.6873</td>\n",
       "      <td>5.7399</td>\n",
       "      <td>-0.6193</td>\n",
       "      <td>3.5645</td>\n",
       "      <td>12.6580</td>\n",
       "      <td>2.5996</td>\n",
       "    </tr>\n",
       "  </tbody>\n",
       "</table>\n",
       "<p>5 rows × 19387 columns</p>\n",
       "</div>"
      ],
      "text/plain": [
       "     Sample_Type Primary_Site  GABARAP  GABARAP_1  GABARAP_2  ENSG00000198888  \\\n",
       "0  Normal Tissue        Blood    11.27      7.823      9.316          11.4278   \n",
       "1  Normal Tissue        Blood    11.24      6.587      9.797          11.1244   \n",
       "2  Normal Tissue        Blood    11.18      6.138      9.749          10.9395   \n",
       "3  Normal Tissue        Blood    11.17      6.901      9.374          10.6507   \n",
       "4  Normal Tissue        Blood    11.08      6.793      9.684          11.3773   \n",
       "\n",
       "   ENSG00000198763  ENSG00000198804  ENSG00000198712  ENSG00000228253  ...  \\\n",
       "0          11.9220          12.4100          13.1565          12.9439  ...   \n",
       "1          10.6001          11.5536          12.0039          11.6778  ...   \n",
       "2          10.4427          11.5286          12.1191          11.9713  ...   \n",
       "3          10.4939          12.1795          11.8817          11.2718  ...   \n",
       "4          10.9368          11.5022          12.1524          12.4101  ...   \n",
       "\n",
       "   ENSG00000160678  ENSG00000160679  ENSG00000143553  ENSG00000214193  \\\n",
       "0           2.1313           5.8726           4.8314           1.0433   \n",
       "1          -1.8314           5.2601           4.2442           0.6050   \n",
       "2          -0.0574           5.7581           5.2419           0.5568   \n",
       "3           0.3907           5.9556           5.6491          -0.8339   \n",
       "4          -0.0877           5.8043           3.8973           0.3573   \n",
       "\n",
       "   ENSG00000196182  ENSG00000181817  ENSG00000116885  ENSG00000116898  \\\n",
       "0           6.1056           6.3116           1.4494           5.0527   \n",
       "1           6.9870           6.0466          -0.5125           4.4894   \n",
       "2           6.2610           6.3896          -0.4921           4.8575   \n",
       "3           6.0381           5.9619          -2.6349           3.6804   \n",
       "4           7.6873           5.7399          -0.6193           3.5645   \n",
       "\n",
       "   ENSG00000119535  ENSG00000142694  \n",
       "0          11.9031           1.8484  \n",
       "1          12.6822           4.4700  \n",
       "2          12.4269           5.0553  \n",
       "3          11.9587           1.0222  \n",
       "4          12.6580           2.5996  \n",
       "\n",
       "[5 rows x 19387 columns]"
      ]
     },
     "execution_count": 22,
     "metadata": {},
     "output_type": "execute_result"
    }
   ],
   "source": [
    "#Preparation of the global dataframe with GABARAP isoform expression AND whole gene expression\n",
    "data = [[ensembl_normal_protcoding, df_GABARAP_Normal], [ensembl_primary_protcoding, df_GABARAP_Primary]]\n",
    "\n",
    "df_final = []\n",
    "\n",
    "for df in data:\n",
    "    #Merge the two df to have GABARAP isoform expression + whole gene expression \n",
    "  df_merged = pandas.merge(df[1], df[0], on= 'sample')\n",
    "\n",
    "  #delete the sample column and select column\n",
    "  df_merged = df_merged.drop(\"sample\", 1)\n",
    "  df_merged = df_merged.drop(df_merged.columns[5:19], axis=1)\n",
    "  \n",
    "  #save in list\n",
    "  df_final.append(df_merged)\n",
    "\n",
    "ensembl_normal_protcoding_GABARAP_expr = df_final[0]\n",
    "ensembl_primary_protcoding_GABARAP_expr = df_final[1]\n",
    "\n",
    "print(ensembl_normal_protcoding_GABARAP_expr.shape)\n",
    "print(ensembl_primary_protcoding_GABARAP_expr.shape)\n",
    "ensembl_normal_protcoding_GABARAP_expr.head()"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 23,
   "metadata": {},
   "outputs": [],
   "source": [
    "#save to csv\n",
    "path = 'GABARAP/dataframes_for_input/'\n",
    "ensembl_normal_protcoding_GABARAP_expr.to_csv(output_file_directory+path+'ensembl_normal_GABARAP_proteinGE.tsv',sep = \"\\t\", index=False)\n",
    "ensembl_primary_protcoding_GABARAP_expr.to_csv(output_file_directory+path+'ensembl_primary_GABARAP_proteinGE.tsv',sep = \"\\t\", index=False)"
   ]
  }
 ],
 "metadata": {
  "colab": {
   "collapsed_sections": [
    "scwtwuDCqUKz",
    "fbjL51C8RltI",
    "R8fLWHXOBYJl"
   ],
   "name": "all_organs.v004.ipynb",
   "provenance": []
  },
  "kernelspec": {
   "display_name": "Python 3",
   "language": "python",
   "name": "python3"
  },
  "language_info": {
   "codemirror_mode": {
    "name": "ipython",
    "version": 3
   },
   "file_extension": ".py",
   "mimetype": "text/x-python",
   "name": "python",
   "nbconvert_exporter": "python",
   "pygments_lexer": "ipython3",
   "version": "3.8.5"
  }
 },
 "nbformat": 4,
 "nbformat_minor": 1
}
